src/log/log_slot.c

```diff
@@ -262,7 +262,7 @@ __wt_log_slot_new(WT_SESSION_IMPL *session)
 		/*
 		 * If we didn't find any free slots signal the worker thread.
 		 */
-		(void)__wt_cond_auto_signal(session, conn->log_wrlsn_cond);
+		WT_RET(__wt_cond_auto_signal(session, conn->log_wrlsn_cond));
```

Comment:
The assumption is that if `wt_log_slot_new` returns an error then `log->active_slot` is not yet ready.  The `WT_MYSLOT_CLOSE` flag is part of the error handling.  So both should be okay because they're returning an error indicating a new active slot is not ready.